test/concurrent/ThreadTest.ooc

```diff
@@ -45,15 +45,17 @@ ThreadTest: class extends Fixture {
 				value set(value get() + 1)
 				Thread yield()
 				Time sleepMilli(1)
+				if (Thread currentThread() isCanceled())
```

Comment:
To be honest I don't like implementing it in the base Thead class and solving this by introducing reference variables or global Thread tables seems kind of hackish. Implement it in the `ThreadTask` abstract class instead and move the responsibility onto the user, this way you will have a clear abstraction layer where we can add more functionallity later. By default the task can be wrapped around a stub object without the cancel functionality so you can even keep the current interface (and behaviour too in most cases, as cancelling does not work on android). It will be quite easy to write a Task subclass with shared mutex for the `TheadPool` too.